redash/handlers/embed.py

```diff
@@ -0,0 +1,47 @@
+import logging
+
+from flask import render_template, request, redirect, session, url_for, flash
+from flask.ext.restful import abort
+
+from flask_login import current_user, login_required
+
+from redash import models, settings
+from redash.wsgi import app
+from redash.utils import json_dumps
+
+@app.route('/embed/query/<query_id>/visualization/<visualization_id>', methods=['GET'])
+@login_required
+def embed(query_id, visualization_id):
+
+    v = models.Visualization.get(models.Visualization.id == visualization_id)
+    qr = {}
+    if v:
+        q = v.query
+        if q.id == int(query_id):
+            v = v.to_dict()
+            query_result_id = q._data['latest_query_data']
+            if query_result_id:
+                qr = models.QueryResult.get_by_id(query_result_id)
+                qr = qr.to_dict()
+            else:
+                abort(400, message="No Results for this query")
+        else:
+            logging.error("%r != %r" % (q.id,query_id))
+            abort(404, message="Invalid visualization for query")
+    else:
+        abort(404, message="Visualization not found.")
+
+    client_config = {
```

Comment:
Let's move `client_config` to somewhere where it will be shared between this function and the index one.